core/src/main/java/org/mule/api/metadata/InvalidExecutableIdException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.metadata;
+
+/**
+ * Represents that the given ExecutableId is invalid due that the Executable was not find
+ * in the desired flow.
+ */
+public class InvalidExecutableIdException extends Exception
```

Comment:
this should be a MuleException. Talk to LG and align this to what he envisions on this component id spec